hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/store/impl/ZookeeperFederationStateStore.java

```diff
@@ -194,12 +205,19 @@ public class ZookeeperFederationStateStore implements FederationStateStore {
   private int maxAppsInStateStore;
 
   /** Directory to store the delegation token data. **/
+  private Map<Integer, String> routerAppRootHierarchies;
   private String routerRMDTSecretManagerRoot;
   private String routerRMDTMasterKeysRootPath;
   private String routerRMDelegationTokensRootPath;
   private String routerRMSequenceNumberPath;
   private String routerRMMasterKeyIdPath;
 
+  private int appIdNodeSplitIndex = 0;
+  private final int HIERARCHIES_LEVEL = 4;
```

Comment:
Add a comment on what the four are. Probably referring to the hierarchy from before but shorter.
`APPLICATION\HIERARCHIES\ID\APP_ID`